### PR TITLE
[5.1] Add slicing notation to F expressions

### PR DESCRIPTION
**Pull request by Viicos**
See https://github.com/django/django/pull/17596/

Feel free to merge this whenever is the right time (django 5.1 isn't released yet)

### CODE REVIEW

File: django-stubs/db/models/expressions.pyi [hunk 1]
```diff
@@ -152,7 +153,10 @@ class F(_Deconstructible, Combinable):
         nulls_first: bool | None = ...,
         nulls_last: bool | None = ...,
     ) -> OrderBy: ...
-    def copy(self) -> F: ...
+    def copy(self) -> Self: ...
+
+class Sliced(F):
+    def __init__(self, obj: F, subscript: int | slice) -> None: ...
```
Comment (Viicos) [inline django-stubs/db/models/expressions.pyi:162]: Todo I need to add `resolve_expression` here

File: django-stubs/db/models/expressions.pyi [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import datetime
 from collections.abc import Callable, Iterable, Iterator, Mapping, Sequence
 from decimal import Decimal
-from typing import Any, ClassVar, Generic, Literal, TypeVar
+from typing import Any, ClassVar, Generic, Literal, NoReturn, TypeVar
```
Comment (sobolevn) [inline django-stubs/db/models/expressions.pyi:4]: Please, use `Never`

Comment (Viicos) [reply django-stubs/db/models/expressions.pyi:4]: Is there a specific reason to do so? I know they are equivalent but I like sticking to `NoReturn` for return values. What do you think?

Comment (sobolevn) [reply django-stubs/db/models/expressions.pyi:4]: OK :)

**Review by sobolevn — APPROVED** [review]

**Pull request merged**